### DISCUSSION TITLE
Add Support for Language Dialect Consistency in Conversations

### DISCUSSION THREAD

**Discussion by salvacarrion** [LAION-AI/Open-Assistant#3371] [Ideas]
Hello,

I’m reaching out to propose a new feature that I believe would enhance the user experience.

### Issue

Currently, when selecting a language, the system does not differentiate between language variants. For instance, when Spanish is selected, dialogues are mixed between European Spanish (Spain) and Latin American Spanish. Similarly, with Catalan, where there is a mix of dialects (Catalan, Valencia, Balearic), and with Portuguese (Brazil, Portugal). This occasionally results in sentences and phrases that, while technically correct, can be perceived as "off" or "weird" by native speakers. I presume this happens with many other languages as well.

### Proposed Solution

I would like to suggest adding a more granular control over the language setting so that we can choose a specific variant (e.g. European Spanish, Mexican Spanish, etc.) for each language. Ideally, once a variant is chosen, the conversation thread should maintain consistency in the use of that variant throughout.

**Suggested Implementation:**
- Include a dropdown or a set of options under the language selection for users to choose the desired variant.
- Store the language variant selection and use it to keep the conversation thread consistent.

### Benefits

- **Enhanced Readability**: Ensuring consistency in language variants makes the conversation more readable and relatable for native speakers.
- **Greater Precision**: Some dialects have unique expressions or terminology. Maintaining consistency in language variants allows for more precise communication.
- **Cultural Sensitivity**: Respecting and utilizing the correct language variant reflects cultural awareness and sensitivity.

**Comment by andreaskoepf** [comment]
It would be great to have a language-variant selection. Could you imagine to start working on this? It would first require some more research how to represent the sub-languages, how to load strings for the UI etc. Currently languages with different language codes are completely separated and appear independently of each other in the language drop down box, e.g. see "Portuguese (Basil)" and "Portuguese".

A challenge of finer-grained language separation is the critical-mass of users required for the crowed-source review process to work. For example a couple of weeks ago we added German dialects Bavarian (Bairisch) and Swabian (Schäbisch) which might be somewhat comparable to Catalan dialects (numbers of users)  - both are stuck in the labeling process of the initial prompts (which requires 3-4 other users additionally to the author to review & approve the message). You can observe similar challenges for other languages with a smaller number of native-speakers.

**Comment by salvacarrion** [comment]
I'd love to work on this but at the moment I have too much work already. If someone can take it on, great. If not, I'll try to work on this when I'm more free.
